### PR TITLE
Added an option to pick a cities database

### PR DESCRIPTION
**Pull request by tkafka**
For my project, I needed as granular location database as possible, so I added an option to pick a specific cities database (one of CITIES_FILES = ['cities500', 'cities1000', 'cities5000', 'cities15000']).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,7 +72,7 @@ geocoder.init(
 ```
 
 Optionally `init()` allows you to specify the directory that geonames files are
-downloaded and cached in.
+downloaded and cached in, and specific the cities database.
```
Comment (tomayac) [inline README.md:75]:
```suggestion
downloaded and cached in, and a specific cities database to be used.
```

File: app.js [hunk 2]
```diff
@@ -59,6 +59,7 @@ app.listen(port, function () {
   );
   geocoder.init(
     {
+      cities_file_override: 'cities500',
```
Comment (tomayac) [inline app.js:62]:
```suggestion
      citiesFileOverride: 'cities500',
```

File: index.js [hunk 3]
```diff
@@ -105,6 +106,8 @@ var geocoder = {
   _admin4Codes: null,
   _alternateNames: null,
 
+  _cities_file_override: null,
```
Comment (tomayac) [inline index.js:109]:
```suggestion
  _citiesFileOverride: null,
```

File: index.js [hunk 4]
```diff
@@ -441,17 +444,18 @@ var geocoder = {
   },
 
   _getGeoNamesCitiesData: function (callback) {
+    var cities_file_overriden = this._cities_file_override || CITIES_FILE
```
Comment (tomayac) [inline index.js:447]:
```suggestion
    var citiesFileOverridden = this._citiesFileOverride || CITIES_FILE
```

File: index.js [hunk 5]
```diff
@@ -441,17 +444,18 @@ var geocoder = {
...
     this._getData(
       // dataName
       'cities',
       // baseName
-      CITIES_FILE,
+      cities_file_overriden,
```
Comment (tomayac) [inline index.js:452]:
```suggestion
      citiesFileOverridden,
```

File: index.js [hunk 6]
```diff
@@ -441,17 +444,18 @@ var geocoder = {
...
       // geonamesZipFilename
-      `${CITIES_FILE}.zip`,
+      `${cities_file_overriden}.zip`,
```
Comment (tomayac) [inline index.js:454]:
```suggestion
      `${citiesFileOverridden}.zip`,
```

File: index.js [hunk 7]
```diff
@@ -441,17 +444,18 @@ var geocoder = {
...
       // fileNameInsideZip
-      `${CITIES_FILE}.txt`,
+      `${cities_file_overriden}.txt`,
```
Comment (tomayac) [inline index.js:456]:
```suggestion
      `${citiesFileOverridden}.txt`,
```

File: index.js [hunk 8]
```diff
@@ -441,17 +444,18 @@ var geocoder = {
...
       // outputFileFolderWithoutSlash
-      GEONAMES_DUMP + '/cities',
+      GEONAMES_DUMP + '/' + cities_file_overriden,
```
Comment (tomayac) [inline index.js:458]:
```suggestion
      GEONAMES_DUMP + '/' + citiesFileOverridden,
```

File: index.js [hunk 9]
```diff
@@ -616,6 +620,12 @@ var geocoder = {
 
     options.load = options.load || {};
 
+    if (CITIES_FILES.indexOf(options.cities_file_override) > -1) {
+      // valid city file
+      this._cities_file_override = options.cities_file_override
+      debug(`Using ${options.cities_file_override} as override of cities database`);
```
Comment (tomayac) [inline index.js:626]:
```suggestion
      this._citiesFileOverride = options.citiesFileOverride;
      debug(`Using ${options.citiesFileOverride} as override of cities database`);
```

**Comment by tomayac** [comment]
Thanks for the PR. Please carefully re-check if everything still works after my changes, which mostly just fix the code conventions. Respond here, and I'm happy to merge.

**Comment by tkafka** [comment]
@tomayac Hi Thomas, I confirm the geocoder works with your changes, good to merge!

**Pull request merged**